akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -488,6 +488,12 @@ object ShardRegion {
       extends ClusterShardingSerializable
       with DeadLetterSuppression
 
+  /**
+   * Updated set of shards sent to all active shards when a shard is started or stopped.
+   * Used for passivation strategies that change limits based on the number of active shards.
+   */
+  final case class ShardsUpdated(shardIds: Set[ShardRegion.ShardId])
```

Comment:
I was originally thinking of handling the remainder when dividing up the limit — where it could sort the shard ids and assign an extra +1 to the first so many based on remainder. But not sure this should be worried about. I've switched to just be the active shard count now.